### PR TITLE
Serialization bugfixes

### PR DESCRIPTION
**Pull request by petersteneteg**

### CODE REVIEW

File: include/inviwo/core/properties/optionproperty.h [hunk 1]
```diff
@@ -193,14 +204,19 @@ class TemplateOptionProperty : public BaseOptionProperty {
...
+    virtual bool isSelectedIdentifier(std::string_view identifier) const override;
+    virtual bool isSelectedDisplayName(std::string_view name) const override;
     bool isSelectedValue(const T& val) const;
 
     const T& get() const;
     const T& operator*() const;
     const T* operator->() const;
-    void set(const T& value);
+
+    bool set(const T& value);
```
Comment (ResearchDaniel) [inline include/inviwo/core/properties/optionproperty.h:231]: Return true if modified?

File: src/core/properties/cameraproperty.cpp [hunk 2]
```diff
@@ -382,16 +366,50 @@ void CameraProperty::addCamerapProperty(std::unique_ptr<Property> camprop) {
...
+}
+
+bool CameraProperty::needsSerialization() const {
+    return serializationMode_ != PropertySerializationMode::None;
+}
+
 void CameraProperty::serialize(Serializer& s) const {
     CompositeProperty::serialize(s);
-    s.serialize("Camera", camera_);
+    if (serializationMode_ != PropertySerializationMode::None) {
```
Comment (ResearchDaniel) [inline src/core/properties/cameraproperty.cpp:397]: Are we doing this check for other properties?

**Review by ResearchDaniel — DISMISSED** [review]

File: include/inviwo/core/properties/optionproperty.h (hunk 1 above)
Comment (petersteneteg) [reply include/inviwo/core/properties/optionproperty.h:231]: it returns true if it was actually set, i.e. if the value was found in the list of options

Comment (petersteneteg) [reply include/inviwo/core/properties/optionproperty.h:231]: will add a comment...

File: src/core/properties/cameraproperty.cpp (hunk 2 above)
Comment (petersteneteg) [reply src/core/properties/cameraproperty.cpp:397]: the value wrapper usually handles that automatically

**Review by martinfalk — DISMISSED** [review]

**Review by ResearchDaniel — DISMISSED** [review]
Not sure if some of the changes also affect the json module. The added functions might need to be reflected there as well

**Comment by petersteneteg** [comment]
@ResearchDaniel @martinfalk poke

**Review by ResearchDaniel — APPROVED** [review]

**Pull request merged**